src/Chapter2.hs

```diff
@@ -349,7 +351,12 @@ Implement a function that returns only the first half of a given list.
 "b"
 -}
 -- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
+firstHalf :: [a] -> [a]
+firstHalf l = go l l
+  where
+    go (h:t) (_:_:t') = h : go t t'
+    go _ _ = []
```

Comment:
Interesting idea! 💡 

You could also use `take` for half of the length of the list 🙂